src/cryptonote_core/service_node_list.cpp

```diff
@@ -315,8 +377,8 @@ namespace service_nodes
         )
       );
       // set the winner as though it was re-registering at transaction index=-1 for this block
-      m_service_nodes_infos[winner_pubkey].block_height = 0;
-      m_service_nodes_infos[winner_pubkey].transaction_index = -1;
+      m_service_nodes_infos[winner_pubkey].last_reward_block_height = block_height;
+      m_service_nodes_infos[winner_pubkey].last_reward_transaction_index = -1;
```

Comment:
> earlier then it should

1. Winner is selected and added to the list.
2 block is processed and new registrants are added to the list.

I don't see the problem with this, it changes almost nothing in the long term to swap them.

Is there something I missed?

Reply:
> Should the transaction index be like however many transactions there are in this block + 1? 

No

Reply:
I think we should fix it so that it is, it's okay but it's technically incorrect and in the worst case that 1 block delay in receiving rewards for other legitimate participants in that block could be the difference between being de-registered or expiring.

Reply:
changed